src/API.jl

```diff
@@ -1493,6 +1495,13 @@ function instantiate(ctx::Context; manifest::Union{Bool, Nothing}=nothing,
         pkgerror("expected manifest file at `$(ctx.env.manifest_file)` but it does not exist")
     end
     Types.check_warn_manifest_julia_version_compat(ctx.env.manifest, ctx.env.manifest_file)
+
+    if Operations.is_manifest_current(ctx.env) === false
+        @warn """The project and manifest may be out of sync.
+            A project dependency has been added/removed or compat entry has changed since the manifest was last resolved.
+            The environment may need to be updated. Try `Pkg.resolve()` or `Pkg.update()`."""
```

Comment:
@KristofferC what about this?
```suggestion
            The environment may need to be updated. Try `Pkg.resolve()` or consider `Pkg.update()`."""
```